src/main/java/org/folio/circulation/resources/ChangeDueDateResource.java

```diff
@@ -0,0 +1,106 @@
+import org.folio.circulation.support.Clients;
+import org.folio.circulation.support.NoContentResult;
+import org.folio.circulation.support.Result;
+import org.folio.circulation.support.RouteRegistration;
+import org.folio.circulation.support.http.server.WebContext;
+import org.joda.time.DateTime;
+
+import io.vertx.core.http.HttpClient;
+import io.vertx.core.json.JsonObject;
+import io.vertx.ext.web.Router;
+import io.vertx.ext.web.RoutingContext;
+
+public class ChangeDueDateResource extends Resource {
+  
+  public ChangeDueDateResource(HttpClient client) {
+    super(client);
+  }
+
+  @Override
+  public void register(Router router) {
+    new RouteRegistration("/circulation/loans/:id/change-due-date", router)
+      .create(this::changeDueDate);
+  }
+
+  private void changeDueDate(RoutingContext routingContext) {
+    createChangeDueDateRequest(routingContext)
+      .after(r -> processChangeDueDate(r, routingContext))
+      .thenApply(NoContentResult::from)
+      .thenAccept(r -> r.writeTo(routingContext.response()));
+  }
+
+  private CompletableFuture<Result<LoanAndRelatedRecords>> processChangeDueDate(
+    final ChangeDueDateRequest request, RoutingContext routingContext) {
+
+    final WebContext context = new WebContext(routingContext);
+    final Clients clients = Clients.create(context, client);
+
+    final LoanRepository loanRepository = new LoanRepository(clients);
+
+    final DueDateScheduledNoticeService scheduledNoticeService
+      = DueDateScheduledNoticeService.using(clients);
+
+    final LoanNoticeSender loanNoticeSender = LoanNoticeSender.using(clients);
+    return succeeded(request)
+      .after(r -> loanRepository.getById(r.getLoanId()))
+      .thenApply(LoanValidator::refuseWhenLoanIsClosed)
+      .thenApply(this::toLoanAndRelatedRecords)
+      .thenApply(ItemStatusValidator::refuseWhenItemIsDeclaredLost)
+      .thenApply(ItemStatusValidator::refuseWhenItemIsClaimedReturned)
+      .thenApply(r -> changeDueDate(r, request))
+      .thenComposeAsync(r -> r.after(loanRepository::updateLoan))
+      .thenApply(r -> r.next(scheduledNoticeService::rescheduleDueDateNotices))
+      .thenCompose(r -> r.after(loanNoticeSender::sendManualDueDateChangeNotice));
+  }
+
+  private Result<LoanAndRelatedRecords> changeDueDate(
+    Result<LoanAndRelatedRecords> loanResult, ChangeDueDateRequest request) {
+
+    return loanResult.map(l -> changeDueDate(l, request.getDueDate()));
+  }
+
+  private LoanAndRelatedRecords changeDueDate(
+    LoanAndRelatedRecords loanAndRelatedRecords, DateTime dueDate) {
+
+    loanAndRelatedRecords.getLoan().changeDueDate(dueDate);
+    return loanAndRelatedRecords;
+  }
+
+  private Result<ChangeDueDateRequest> createChangeDueDateRequest(
+    RoutingContext routingContext) {
+
+    final String loanId = routingContext.pathParam("id");
+    final JsonObject body = routingContext.getBodyAsJson();
+
+    if (!body.containsKey(ChangeDueDateRequest.DUE_DATE)) {
```

Comment:
Personally, I think statically importing the `DUE_DATE` constant could improve the readability of this line